tagger.go

```diff
@@ -65,85 +69,240 @@ func newLinodeClient() linodego.Client {
+
+		var combinedNewTags, tags []string
+		var linodeID int
+
+		switch objects := objects.(type) {
+		case []linodego.Instance:
+
+			instanceType := "linodes"
+			instanceTagMap[instanceType] = make(map[int][]string)
+
+			if linodeTagSlice[instanceType] != nil {
+				for _, linode := range objects {
+
+					combinedNewTags, tags, linodeID = compareTags(linode, linodeTagSlice[instanceType])
+					instanceTagMap[instanceType][linodeID] = combinedNewTags
+					logTags(combinedNewTags, tags, linodeID)
+				}
+			}
+		case []linodego.Volume:
+
+			instanceType := "volumes"
+			instanceTagMap[instanceType] = make(map[int][]string)
 
-		if len(combinedNewTags) > 0 {
-			linodeIDTagMap[linode.ID] = combinedNewTags
-			sort.Strings(combinedNewTags)
-			if !slices.Equal(tags, combinedNewTags) {
-				log.WithFields(log.Fields{
-					"linode_id": linode.ID,
-					"old_tags":  tags,
-					"new_tags":  combinedNewTags,
-				}).Debug("Linode tag set updated")
+			if linodeTagSlice[instanceType] != nil {
+				for _, volume := range objects {
+					combinedNewTags, tags, linodeID = compareTags(volume, linodeTagSlice[instanceType])
+					instanceTagMap[instanceType][linodeID] = combinedNewTags
+					logTags(combinedNewTags, tags, linodeID)
+				}
+			}
+		case []linodego.NodeBalancer:
+
+			instanceType := "nodebalancers"
+			instanceTagMap[instanceType] = make(map[int][]string)
+
+			if linodeTagSlice[instanceType] != nil {
+				for _, nodebalancer := range objects {
+					combinedNewTags, tags, linodeID = compareTags(nodebalancer, linodeTagSlice[instanceType])
+					instanceTagMap[instanceType][linodeID] = combinedNewTags
+					logTags(combinedNewTags, tags, linodeID)
+				}
+			}
+		case []linodego.Domain:
+
+			instanceType := "domains"
+			instanceTagMap[instanceType] = make(map[int][]string)
+
+			if linodeTagSlice[instanceType] != nil {
+				for _, domain := range objects {
+					combinedNewTags, tags, linodeID = compareTags(domain, linodeTagSlice[instanceType])
+					instanceTagMap[instanceType][linodeID] = combinedNewTags
+					logTags(combinedNewTags, tags, linodeID)
+				}
+			}
+		case []linodego.LKECluster:
+
+			instanceType := "lkeclusters"
+			instanceTagMap[instanceType] = make(map[int][]string)
+
+			if linodeTagSlice[instanceType] != nil {
+				for _, lkecluster := range objects {
+					combinedNewTags, tags, linodeID = compareTags(lkecluster, linodeTagSlice[instanceType])
+					instanceTagMap[instanceType][linodeID] = combinedNewTags
+					logTags(combinedNewTags, tags, linodeID)
+				}
 			}
 		}
+
 	}
 
-	return linodeIDTagMap, nil
+	return instanceTagMap, nil
 }
 
-func updateLinodeInstanceTags(ctx context.Context, client linodego.Client, id int, tags *[]string) error {
-	updatedInstance, err := client.UpdateInstance(ctx, id, linodego.InstanceUpdateOptions{Tags: tags})
-	if err != nil {
-		return err
-	}
+func updateLinodeInstanceTags(ctx context.Context, client linodego.Client, id int, tags *[]string, data string) error {
```

Comment:
```suggestion
func updateObjectTags(ctx context.Context, client linodego.Client, id int, tags *[]string, data string) error {
```

I feel this can benefit from an updated name as well